lib/core/test/unit/Cardano/Wallet/Primitive/TypesSpec.hs

```diff
@@ -362,6 +362,14 @@ spec = do
                             . slotStartTime sps
                     Just slot === f slot
 
+        it "slotStartTime (slotRange timeRange) ⊆ timeRange " $
+            withMaxSuccess 1000 $ property $
+                \sps timeRange -> do
+                    let r = getUniformTime <$> timeRange
+                    case fmap (slotStartTime sps) <$> slotRange sps r of
+                        Just r' -> r' `rangeIsSubrangeOf` r
```

Comment:
Do you think we should add some coverage analysis here to ensure that we cover both the `Just` and the `Nothing` cases?

We can use `cover` and `checkCoverage` to do this. (See `"arbitrary ranges are valid"` for an example.)

Reply:
:+1: with some restructuring it was not that awkward to do

I don't think we have to cover the `Nothing` case though.

Reply:
> +1 with some restructuring it was not that awkward to do
> 
> I don't think we care about covering the `Nothing` case though.

Function `slotRange` should only produce `Nothing` when the upper bound is earlier than the start of the blockchain. So if we get a `Nothing`, we should check that this condition is satisfied.

Reply:
In my eyes the `"f timeRange ⊆ timeRange"` doesn't care about _why_ or _when_ `slotRange` returns `Nothing`, which is tested elsewhere

Reply:
Agree that this is tested elsewhere. :+1: